packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -565,6 +566,52 @@ class MediaQueryData {
     );
   }
 
+  /// Creates a copy of this media query data by removing [displayFeatures] that
+  /// are completely outside the given sub-screen and adjusting the [padding],
+  /// [viewInsets] and [viewPadding] to be zero on the sides that are not
+  /// included in the sub-screen.
+  ///
+  /// Returns unmodified [MediaQueryData] if the sub-screen coincides with the
+  /// available screen space.
+  ///
+  /// Throws if the given sub-screen is outside the available screen space.
+  ///
+  /// See also:
+  ///
+  ///  * [DisplayFeatureSubScreen], which removes the display features that
+  ///  split the screen, from the [MediaQuery] and adds a [Padding] widget to
+  ///  position the child to match the selected sub-screen.
+  MediaQueryData removeDisplayFeatures(Rect subScreen) {
+    assert(subScreen.left >= 0.0 && subScreen.top >= 0.0 &&
```

Comment:
Can you add an error message to this assertion to help folks understand what is wrong?

Reply:
Yes, done.